clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,8 +497,22 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
-
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
+        new ApiExceptionBuilder() {
+            @Override
+            public ApiException build(String message) {
+                return new KafkaStorageException(message);
+            }
+    }),
+    UNKNOWN_ERROR_CODE(57, "The client received an unknown error code when processing the response. This may happen if " +
+            "the client library version is lower than the server library version",
+        new ApiExceptionBuilder() {
```

Comment:
@junrao My use-case for UnknownErrorCodeException is based on the assumption that it is always OK for client to retry. The client should always have a reasonable timeout setting if it retries. Thus if the original error is non-retriable, the client will block unnecessarily up to that timeout which is not that bad. Also, for requests that don't want to retry on unknown error code, we can always update that its response handling logic to skip retry if error == UnknownErrorCodeException.

Does this sound reasonable? If not, then I think we will have to add specific if/else potentially a few requests for every new error in the future. Or I can remove the UnknownErrorCodeException if you prefer not to make the decision in this patch.